code/modules/modular_computers/file_system/programs/generic/folding.dm

```diff
@@ -0,0 +1,119 @@
+
+#define MINIMUM_SCIENCE_INTERVAL 450
+#define MAXIMUM_SCIENCE_INTERVAL 900
+#define MINIMUM_FOLDING_EVENT_INTERVAL 50
+#define MAXIMUM_FOLDING_EVENT_INTERVAL 100
+#define SCIENCE_MONEY_PER_MINUTE 0.08 // So little money.
+
+/datum/computer_file/program/folding
+	filename = "fldng"
+	filedesc = "FOLDING@SAPCE"
+	extended_desc = "This program uses processor cycles for science, in exchange for money."
+	program_icon_state = "generic"
+	program_key_state = "generic_key"
+	size = 6
+	available_on_network = 1
+	usage_flags = PROGRAM_ALL
+	nanomodule_path = /datum/nano_module/program/folding
+	category = PROG_UTIL
+	requires_network = 1
+
+	var/started_on 			= 0 	// When the program started some science.
+	var/current_interval 	= 0		// How long the current interval will be.
+	var/next_event 			= 0 	// in world timeofday, when the next event is scheduled to pop.
+	var/crashed				= FALSE	// Program periodically needs a restart.
+	var/crashed_at			= 0		// When the program crashed.
+
+/datum/computer_file/program/folding/Topic(href, href_list)
+	if(..())
+		return TOPIC_HANDLED
+	. = TOPIC_REFRESH
+
+	if(href_list["fix_crash"] && crashed)
+		current_interval += world.timeofday - crashed_at
+		crashed = FALSE
+
+	if(href_list["start"] && started_on == 0)
+		started_on = world.timeofday
+		current_interval = rand(MINIMUM_SCIENCE_INTERVAL, MAXIMUM_SCIENCE_INTERVAL) SECONDS
+		// Better processors can do this faster. Normal processors are slower.
+		var/obj/item/stock_parts/computer/processor_unit/processor = computer.get_component(PART_CPU)
+		current_interval = (processor.processing_power / 2) * current_interval
+		next_event = (rand(MINIMUM_FOLDING_EVENT_INTERVAL, MAXIMUM_FOLDING_EVENT_INTERVAL) SECONDS) + world.timeofday
+
+	if(href_list["collect"] && started_on > 0 && !crashed)
+		var/obj/item/card/id/I = usr.GetIdCard()
+		if(!I)
+			to_chat(usr, SPAN_WARNING("Unable to locate ID card for transaction."))
+			return TOPIC_HANDLED
+		var/datum/money_account/account = get_account(I.associated_account_number)
+		var/earned = current_interval * SCIENCE_MONEY_PER_MINUTE
+		account.deposit(earned, "Completed FOLDING@SPACE project.")
+		to_chat(usr, SPAN_NOTICE("Transferred [earned] to your account."))
+		started_on = 0
+		current_interval = 0
+
+	if(.)
+		SSnano.update_uis(NM)
+
+/datum/computer_file/program/folding/process_tick()
+	. = ..()
+	if(!started_on)
+		return
+
+	if(world.timeofday > next_event)
+		return
+
+	var/mob/living/holder = get_holder_of_type(computer, /mob/living/carbon/human)
+	if(!crashed)
+		if(holder)
+			switch(rand(1,3))
+				if(1)
+					to_chat(holder, SPAN_WARNING("The [computer] starts to get very warm."))
+				if(2)
+					to_chat(holder, SPAN_WARNING("The [computer] gets scaldingly hot, burning you!"))
+					holder.burn_skin(0.45)
+				if(3)
+					to_chat(holder, SPAN_WARNING("The [computer] pings an error chime."))
+					crashed = TRUE
+					crashed_at = world.timeofday
+		else
+			crashed = TRUE
+			crashed_at = world.timeofday
```

Comment:
So, it *always* crashes if you're not holding it and only sometimes crashes if you are?

Reply:
Yes. It's the balance of trying to escape consequences by putting it down.